src/trees/bp/mod.rs

```diff
@@ -340,6 +347,29 @@ impl<const BLOCK_SIZE: usize> BpTree<BLOCK_SIZE> {
         debug_assert!(index < self.vec.len(), "Index out of bounds");
         self.vec.rank1(index + 1) as i64 - self.vec.rank0(index + 1) as i64
     }
+
+    /// Iterate over the nodes of the tree in depth-first (pre-)order.
+    /// This is the most efficient way to iterate over all nodes of the tree.
+    ///
+    /// If the tree is unbalanced, the iterator returns the node handles in the order they appear in
+    /// the parenthesis expression, and it will return handles that don't have a matching closing
+    /// parenthesis.
+    pub fn dfs_iter(
+        &self,
+    ) -> impl Iterator<Item = <BpTree<BLOCK_SIZE> as Tree>::NodeHandle> + use<'_, BLOCK_SIZE> {
+        self.vec.iter1()
```

Comment:
Wow, I hadn't realized this is possible! I've been using first-child and next-sibling to iterate, but this causes me to rethink my approach.